src/PhpSpec/Runner/Maintainer/CollaboratorsMaintainer.php

```diff
@@ -140,6 +130,9 @@ private function generateCollaborators(CollaboratorManager $collaborators, \Refl
                 if (($indexedClass = $this->getParameterTypeFromIndex($classRefl, $parameter))
                     || ($indexedClass = $this->getParameterTypeFromReflection($parameter))) {
                     $collaborator->beADoubleOf($indexedClass);
+                    if (null !== $indexedClass) {
```

Comment:
@stof This is not strictly related to this PR, but in my opinion, assignments shouldn't be made inside a condition. First of all to avoid mistakes (like doing an assignment when in fact you just want to compare). But also, and this becomes apparent in this code snippet, because of the way, PHP evaluates the value. It is not an identity comparison to false (===) - the 'falsy' part in your description. So this will also evaluate to false for empty strings, or the integer 0 or whatever else PHP is considering 'falsy'. So if you wanted to keep the assignment inside the condition, you should at least do it like that:
`null !== ($foo = $this->bar())`

This is not meant to be a fundamental discussion, I am just interested in your thoughts about that. Is it done like that on purpose here, or would you agree that this is kind of a risky way of writing conditions?